osgp/protocol-adapter-dlms/osgp-protocol-adapter-dlms/src/test/java/org/opensmartgridplatform/adapter/protocol/dlms/application/services/SystemEventServiceTest.java

```diff
@@ -0,0 +1,151 @@
+  }
+
+  @Test
+  void verifyMaxValueNotReachedEvent() {
+    final DlmsDevice device =
+        new DlmsDeviceBuilder()
+            .withInvocationCounter(this.invocationCounterEventThreshold - 1)
+            .build();
+
+    final MessageMetadata messageMetadata =
+        new Builder().withOrganisationIdentification("org-id").build();
+
+    this.service.verifySystemEventThresholdReachedEvent(device, messageMetadata);
+
+    verifyNoInteractions(this.correlationIdProviderService);
+    verify(this.osgpRequestMessageSender, never())
+        .send(
+            any(RequestMessage.class),
+            eq(MessageType.SYSTEM_EVENT.name()),
+            any(MessageMetadata.class));
+  }
+
+  @Test
+  void verifyInvocationCounterIsNull() {
+    final DlmsDevice device = new DlmsDeviceBuilder().withInvocationCounter(null).build();
+
+    final MessageMetadata messageMetadata =
+        new Builder().withOrganisationIdentification("org-id").build();
+
+    this.service.verifySystemEventThresholdReachedEvent(device, messageMetadata);
+
+    verifyNoInteractions(this.correlationIdProviderService);
+    verify(this.osgpRequestMessageSender, never())
+        .send(
+            any(RequestMessage.class),
+            eq(MessageType.SYSTEM_EVENT.name()),
+            any(MessageMetadata.class));
+  }
+
+  @Test
+  void verifyMaxValueReachedEvent() {
+    final DlmsDevice device =
+        new DlmsDeviceBuilder()
+            .withDeviceIdentification("device-1")
+            .withInvocationCounter(this.invocationCounterEventThreshold)
+            .build();
+
+    final MessageMetadata messageMetadata =
+        new Builder()
+            .withOrganisationIdentification("org-id")
+            .withDomain("domain")
+            .withDomainVersion("1.0")
+            .build();
+
+    when(this.correlationIdProviderService.getCorrelationId(
+            messageMetadata.getOrganisationIdentification(), device.getDeviceIdentification()))
+        .thenReturn("corr-id");
+
+    this.service.verifySystemEventThresholdReachedEvent(device, messageMetadata);
+
+    final ArgumentCaptor<MessageMetadata> messageMetadataCaptor =
+        ArgumentCaptor.forClass(MessageMetadata.class);
+    final ArgumentCaptor<RequestMessage> requestMessageCaptor =
+        ArgumentCaptor.forClass(RequestMessage.class);
+
+    verify(this.osgpRequestMessageSender)
+        .send(
+            requestMessageCaptor.capture(),
+            eq(MessageType.SYSTEM_EVENT.name()),
+            messageMetadataCaptor.capture());
+
+    final RequestMessage requestMessage = requestMessageCaptor.getValue();
+    assertThat(requestMessage.getDeviceIdentification())
+        .isEqualTo(device.getDeviceIdentification());
+    assertThat(requestMessage.getDeviceIdentification())
+        .isEqualTo(device.getDeviceIdentification());
+    assertThat(requestMessage.getCorrelationUid()).isEqualTo("corr-id");
+    assertThat(requestMessage.getOrganisationIdentification()).isEqualTo("no-organisation");
```

Comment:
Shouldn't this expect the organization from the original `messageMetadata`?
If it is passed on with 'no-organisation' how is the notification for the event going to end up with the correct organization/application? (See for instance the `DefaultNotificationService` in osgp-adapter-ws-shared, and its use of an `ApplicationDataLookupKey` based on the identification of the organization.)

Reply:
Changed, getOrganisationIdentification copied from metadata
Als ip address is copied now